js/stores/message_store.js

```diff
@@ -23,15 +25,15 @@ var MessageStore = assign({}, EventEmitter.prototype, {
 });
 
 MessageStore.dispatchToken = AppDispatcher.register(function(action) {
-  if (action.type === "NewMessage") {
+  if (action.type === ActionTypes.NEW_MESSAGE) {
     message = {
-      id: MessageStore.all().length + 1,
-      timestamp: new Date().getTime(),
-      user: 'jack',
-      text: action.text
-    }
+      id:  action.data.message_id,
+      timestamp: action.data.timestamp,
+      user: action.data.user,
+      text: action.data.text
+    };
```

Comment:
Fixed the var. Also changed the id. It's kind of weird right now because probably we should be getting the message id from some kind of central storage. But for now anyway its differerent from var message = action.data